src/SchemaRegistry.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.21;
+
+import { Schema } from "./struct/Schema.sol";
+
+contract SchemaRegistry {
+  mapping(bytes32 id => Schema schema) public schemas;
+  uint256 public numberOfSchemas = 0;
+
+  error SchemaAlreadyExists();
+  error SchemaNameMissing();
+  error SchemaStringMissing();
+
+  event SchemaCreated(string name, bytes32 schemaId);
+
+  function getIdFromSchemaString(string calldata _schema) public pure returns (bytes32) {
```

Comment:
we have to decide if underscore prefixes args:
1/ all the time
2/ only to prevent collision between args and variable names inside the method
3/ never (but what do we do?)

This should be documented in the WoW

Reply:
I'd suggest we go with the OpenZeppelin guidelines: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/GUIDELINES.md#solidity-conventions
Meaning we keep the `_` for private and internal functions/variables.